src/collective/documentgenerator/tests/test_pod_template.py

```diff
@@ -175,3 +181,22 @@ def test_template_has_been_modified_method(self):
 
     def test_get_available_formats(self):
         self.assertEqual(self.test_podtemplate.get_available_formats(), ['odt', ])
+
+
+class TestPODTemplateValidator(ConfigurablePODTemplateIntegrationTest):
+
+    def test_file_extension_is_odt(self):
+        pod_template = self.test_podtemplate
+        extension = pod_template.odt_file.filename.split('.')[-1]
+        self.assertEqual(extension, 'odt')
+
+    def test_add_bad_formats_and_get_errormessage(self):
+        pod_template = self.test_podtemplate
+        pod_template.pod_formats.append('xls')
+        view = pod_template.restrictedTraverse('edit')
+        view.form_instance.update()
+        validator = PodFormatsValidator(pod_template, pod_template.REQUEST, view.form_instance, IConfigurablePODTemplate['pod_formats'], pod_template.widget)
```

Comment:
Too long line, translate the msg instead of rebuilding it so you test translation and it does not break if translation changes